testsuites/listener/shared/client_sg/conftest.py

```diff
@@ -119,10 +121,17 @@ def setup_client_syncgateway_test(request, setup_client_syncgateway_suite):
     }
 
     log_info("Tearing down test")
-    client.delete_databases(ls_url)
-    liteserv.stop()
 
-    # if the test failed pull logs
-    if request.node.rep_call.failed:
-        logging_helper = Logging()
-        logging_helper.fetch_and_analyze_logs(cluster_config=cluster_config, test_name=test_name)
+    try:
+        client.delete_databases(ls_url)
+    except HTTPError as he:
+        liteserv.stop()
+        raise HTTPError(he)
+    except:
+        liteserv.stop()
+        raise HTTPError(sys.exc_info()[0])
+    finally:
+        # if the test failed pull logs
+        if request.node.rep_call.failed:
```

Comment:
If an exception in throw in this fixture, are you sure this will be true? One other option is to take this out of the finally and make sure to pull after to stoping the liteserv in the exception handler. I think that may be simpler